src/main/java/com/impactupgrade/integration/sfdc/SFDCPartnerAPIClient.java

```diff
@@ -674,8 +675,19 @@ private QueryResult _query(int count, String queryString) throws ConnectionExcep
     try {
       return partnerConnection.get().query(queryString);
     } catch (ApiFault e) {
-      log.error("query failed due to {}: {}", e.getExceptionCode(), e.getExceptionMessage(), e);
-      throw e;
+      if (e instanceof UnexpectedErrorFault && e.getExceptionMessage().contains("Session timed out")) {
```

Comment:
@VSydor Would it be better to instead look for the "INVALID_SESSION_ID" part of the message? Not sure if there are different types of "time out" conditions.